lib/cdo/aws/s3_packaging.rb

```diff
@@ -0,0 +1,143 @@
+require 'aws-sdk'
+
+class S3Packaging
+  BUCKET_NAME = 'cdo-build-package'
+
+  def initialize(package_name, source_location, target_location)
+    @client = Aws::S3::Client.new
```

Comment:
We should raise an exception in the case where any of the provided arguments are nil.